pytorch_lightning/accelerators/accelerator.py

```diff
@@ -209,6 +215,22 @@ def init_ddp_connection(
                 torch_backend, rank=global_rank, world_size=world_size
             )
 
+    def sync_tensor(self,
+                    tensor: Union[torch.Tensor],
+                    group: Optional[Any] = None,
+                    reduce_op: Optional[Union[ReduceOp, str]] = None) -> torch.Tensor:
+        """
+        Function to reduce a tensor from several distributed processes to one aggregated tensor.
+        Args:
+            tensor: the tensor to sync and reduce
+            group: the process group to gather results from. Defaults to all processes (world)
+            reduce_op: the reduction operation. Defaults to sum.
+                Can also be a string of 'avg', 'mean' to calculate the mean during reduction.
+        Return:
+            reduced value
+        """
+        raise NotImplementedError()
```

Comment:
Why not implemented when in all sub-classes we just import and use a single function...
`from pytorch_lightning.utilities.distributed import sync_ddp_if_available`

Reply:
Not sure I understand the suggestion, can you elaborate?  If we use `sync_ddp_if_available`, then this will not work for Horovod, TPU, etc.

Reply:
I understand the weirdness here, it's a bit weird to have `sync_ddp_if_available` within the base accelerator class, but as a side effect we see it implemented in many ddp based implementations. I think the right thing to do would be for the ddp extension accelerators to inherit from ddp, but this is out of the scope of this PR.

Reply:
In general I agree that all the DDP accelerators should share a common base class.  Right now there is a lot of duplication even beyond this method.  However, I didn't want to undertake such a major refactor in this PR.